### ISSUE TITLE
Monitor traffic portchannels

### ISSUE THREAD

**Issue by trentasis** [centreon/centreon-plugins#1977]
Hi,

I'm using interfaces mode to monitor traffic usage on DELL OS10 and alcatel interfaces and works great, but now I'm trying to monitor portchannel (LACP) that exist  but I can't It seems that is not working: for example in DELL OS10:

List-interfaces show potcahennel as unknown:
 perl centreon_plugins.pl --hostname=10.10.10.10 --plugin=os::linux::snmp::plugin --snmp-version=2c --snmp-community=com --mode=list-interfaces
List interfaces:
'mgmt1/1/1' [speed = 1000, status = up, id = 10]
'ethernet1/1/1' [speed = 1000, status = up, id = 12]
'ethernet1/1/2' [speed = , status = down, id = 13]
'ethernet1/1/3' [speed = , status = down, id = 14]
'ethernet1/1/4' [speed = , status = down, id = 15]
'ethernet1/1/5' [speed = , status = down, id = 16]
'ethernet1/1/6' [speed = , status = down, id = 17]
'ethernet1/1/7' [speed = 1000, status = up, id = 18]
'ethernet1/1/8' [speed = 1000, status = up, id = 19]
'ethernet1/1/9' [speed = 1000, status = up, id = 20]
'ethernet1/1/10' [speed = 1000, status = up, id = 21]
'ethernet1/1/11' [speed = 1000, status = up, id = 22]
'ethernet1/1/12' [speed = 1000, status = up, id = 23]
'ethernet1/1/13' [speed = 1000, status = up, id = 24]
'ethernet1/1/14' [speed = 1000, status = up, id = 25]
'ethernet1/1/15' [speed = 1000, status = up, id = 26]
'ethernet1/1/16' [speed = , status = down, id = 27]
'ethernet1/1/17' [speed = , status = down, id = 28]
'ethernet1/1/18' [speed = , status = down, id = 29]
'ethernet1/1/19' [speed = , status = down, id = 30]
'ethernet1/1/20' [speed = , status = down, id = 31]
'ethernet1/1/21' [speed = 10000, status = up, id = 32]
'ethernet1/1/22' [speed = 10000, status = up, id = 33]
'ethernet1/1/23' [speed = 10000, status = up, id = 34]
'ethernet1/1/24' [speed = 10000, status = up, id = 35]
'ethernet1/1/25' [speed = 100000, status = up, id = 36]
'ethernet1/1/26' [speed = 100000, status = up, id = 37]
'ethernet1/1/27' [speed = , status = down, id = 38]
'ethernet1/1/28' [speed = , status = down, id = 39]
'ethernet1/1/29' [speed = , status = down, id = 40]
'ethernet1/1/30' [speed = 10000, status = up, id = 41]
'port-channel30' [speed = 10000, status = unknown, id = 42]
'port-channel4' [speed = , status = unknown, id = 44]
'port-channel5' [speed = 1000, status = unknown, id = 46]
'vlan1' [speed = 10000, status = up, id = 48]
'vlan2' [speed = 10000, status = up, id = 57]
'vlan400' [speed = 10000, status = up, id = 58]
'vlan401' [speed = 10000, status = up, id = 65]
'vlan402' [speed = 10000, status = up, id = 74]
'vlan4094' [speed = 10000, status = up, id = 81]
'port-channel1000' [speed = 200000, status = unknown, id = 82]
'port-channel3' [speed = 1000, status = unknown, id = 89]

and using interfaces 41 works:
 perl centreon_plugins.pl --hostname=10.10.10.10 --plugin=os::linux::snmp::plugin --snmp-version=2c --snmp-community=com --mode=interfaces --statefile-dir=/tmp/centreonTMP/  --add-traffic --nagvis-perfdata  --add-errors  --add-speed --interface='41'  --warning-in-traffic=70 --warning-out-traffic=70 --critical-in-traffic=90 --critical-out-traffic=90
OK: Interface 'ethernet1/1/30' Traffic In : skipped, Traffic Out : skipped, Packets In Discard : skipped, Packets In Error : skipped, Packets Out Discard : skipped, Packets Out Error : skipped, Speed : 10.00Gb/s | 'speed'=10000000000b/s;;;0;

but 42 that is a portchannel not:
 perl centreon_plugins.pl --hostname=10.10.10.10 --plugin=os::linux::snmp::plugin --snmp-version=2c --snmp-community=com --mode=interfaces --statefile-dir=/tmp/centreonTMP/  --add-traffic --nagvis-perfdata  --add-errors  --add-speed --interface='42'  --warning-in-traffic=70 --warning-out-traffic=70 --critical-in-traffic=90 --critical-out-traffic=90
OK: Interface 'port-channel30' Traffic In : skipped, Traffic Out : skipped, Packets In Discard : skipped, Packets In Error : skipped, Packets Out Discard : skipped, Packets Out Error : skipped, Speed : 10.00Gb/s | 'speed'=10000000000b/s;;;0;

Same occurs with alcatel omniswitchs

Any suggestion where is the issue?

**Comment by garnier-quentin** [comment]
Maybe that kind of aggregate port (with lacp) is not managed by the standard mib.

**Comment by garnier-quentin** [comment]
Maybe some specific mibs to check for that:
https://docs.paloaltonetworks.com/pan-os/8-1/pan-os-admin/monitoring/snmp-monitoring-and-traps/supported-mibs/ieee-8023-lag-mib

**Comment by garnier-quentin** [comment]
Do you have some informations on that branch ?
.1.2.840.10006.300.43

**Comment by trentasis** [comment]
hi,

Correct, is an aggregate port with lacp, I have sended by email snmp info about this branch

**Comment by garnier-quentin** [comment]
We could create a standard mode for that.

**Comment by trentasis** [comment]
Thanks, It will be a great feature!. About error in list-interface  I understand that unknown error is the same, right?

**Comment by garnier-quentin** [comment]
You should add --add-status to see the status.

**Comment by trentasis** [comment]
yes. but issue is that portchannel al returning unknown for example:
'port-channel1000' [speed = 200000, status = unknown, id = 82]
'port-channel3' [speed = 1000, status = unknown, id = 89]

**Comment by garnier-quentin** [comment]
i think it's by design. you should exclude it and use a new mode (it doesn't exist yet)

**Comment by trentasis** [comment]
Perfect, It will be a great feature, It will be possible?
thanks

**Comment by garnier-quentin** [comment]
It can. but when ?  I don't know

**Comment by garnier-quentin** [comment]
I also need following OIDs:
.1.3.6.1.2.1.2.2.1
1.3.6.1.2.1.31.1.1.1

**Comment by fmattesct** [comment]
Hi,
Thanks for your interest in Centreon.
Requests for new features and enhancements must be suggested [here](https://thewatch.centreon.com/ideas).
Troubleshooting and questions must now be asked [here](https://thewatch.centreon.com/data-collection-6) (cf our [new issue template](https://github.com/centreon/centreon-plugins/blob/develop/.github/issue_template.md)).

If this need is still relevant, can you create a new request on The Watch and include all useful data to help us to prioritize ?

Thank you for your understanding.
